ibis/backends/pandas/executor/generic.py

```diff
@@ -0,0 +1,319 @@
+        value = pd.Timestamp(value, tz="UTC").tz_localize(None)
+
+    return value
+
+
+@execute.register(ops.Field)
+def execute_field(op, rel, name):
+    return rel[name]
+
+
+@execute.register(ops.Alias)
+def execute_alias(op, arg, name):
+    try:
+        return arg.rename(name)
+    except AttributeError:
+        return arg
+
+
+@execute.register(ops.SortKey)
+def execute_sort_key(op, expr, ascending):
+    return expr
+
+
+@execute.register(ops.Not)
+def execute_not(op, arg):
+    if isinstance(arg, (bool, np.bool_)):
+        return not arg
+    else:
+        return ~arg
+
+
+@execute.register(ops.Negate)
+def execute_negate(op, arg):
+    if isinstance(arg, (bool, np.bool_)):
+        return not arg
+    else:
+        return -arg
+
+
+@execute.register(ops.Cast)
+def execute_cast(op, arg, to):
+    if isinstance(arg, pd.Series):
+        return PandasConverter.convert_column(arg, to)
+    else:
+        return PandasConverter.convert_scalar(arg, to)
+
+
+@execute.register(ops.E)
+def execute_e(op):
+    return np.e
+
+
+@execute.register(ops.Pi)
+def execute_pi(op):
+    return np.pi
+
+
+_unary_operations = {
+    ops.Abs: abs,
+    ops.Ceil: lambda x: np.ceil(x).astype("int64"),
+    ops.Floor: lambda x: np.floor(x).astype("int64"),
+    ops.Sign: np.sign,
+    ops.Tan: np.tan,
+    ops.Cos: np.cos,
+    ops.Cot: lambda x: 1 / np.tan(x),
+    ops.Sin: np.sin,
+    ops.Atan: np.arctan,
+    ops.Acos: np.arccos,
+    ops.Asin: np.arcsin,
+    ops.BitwiseNot: np.invert,
+    ops.Radians: np.radians,
+    ops.Degrees: np.degrees,
+}
+
+_binary_operations = {
+    ops.Greater: operator.gt,
+    ops.Less: operator.lt,
+    ops.LessEqual: operator.le,
+    ops.GreaterEqual: operator.ge,
+    ops.Equals: operator.eq,
+    ops.NotEquals: operator.ne,
+    ops.And: operator.and_,
+    ops.Or: operator.or_,
+    ops.Xor: operator.xor,
+    ops.Add: operator.add,
+    ops.Subtract: operator.sub,
+    ops.Multiply: operator.mul,
+    ops.Divide: operator.truediv,
+    ops.FloorDivide: operator.floordiv,
+    ops.Modulus: operator.mod,
+    ops.Power: operator.pow,
+    ops.IdenticalTo: lambda x, y: (x == y) | (pd.isnull(x) & pd.isnull(y)),
+    ops.BitwiseXor: lambda x, y: np.bitwise_xor(x, y),
+    ops.BitwiseOr: lambda x, y: np.bitwise_or(x, y),
+    ops.BitwiseAnd: lambda x, y: np.bitwise_and(x, y),
+    ops.BitwiseLeftShift: lambda x, y: np.left_shift(x, y).astype("int64"),
+    ops.BitwiseRightShift: lambda x, y: np.right_shift(x, y).astype("int64"),
+    ops.Atan2: np.arctan2,
+}
+
+
+@execute.register(ops.Unary)
+def execute_unary(op, arg):
+    return _unary_operations[type(op)](arg)
+
+
+@execute.register(ops.Binary)
+def execute_equals(op, left, right):
+    return _binary_operations[type(op)](left, right)
+
+
+def mapdecimal(func, s):
+    def wrapper(x):
+        try:
+            return func(x)
+        except decimal.InvalidOperation:
+            return decimal.Decimal("NaN")
+
+    return s.map(wrapper)
```

Comment:
```suggestion
    return s.map(wrapper, na_action="ignore"))
```

To ensure that NULL values are preserved and also to avoid wasting cycles on NULL handling.

Reply:
Updated everywhere.